include/SFML/Window/ContextSettings.hpp

```diff
@@ -94,6 +109,23 @@ struct ContextSettings
 /// all handled the same way (i.e. you can use any version
 /// < 3.0 if you don't want an OpenGL 3 context).
 ///
+/// When requesting a context with a version greater or equal
+/// to 3.2, you have the option of specifying whether the
+/// context should follow the core or compatibility profile
+/// of all newer (>= 3.2) OpenGL specifications. For versions
+/// 3.0 and 3.1 there is only the core profile. By default
+/// a compatibility context is created. You only need to specify
+/// the core flag if you want a core profile context to use with
+/// your own OpenGL rendering.
+/// <b>Warning: The graphics module will not function if you
+/// request a core profile context. Make sure the attributes are
+/// set to Default if you want to use the graphics module.</b>
+///
+/// Setting the debug attribute flag will request a context with
+/// additional debugging features enabled. Depending on the
+/// system, this might be required for advanced OpenGL debugging.
+/// OpenGL debugging is disabled by default.
+///
```

Comment:
Regarding core/compatiliby flag with version >= 3.2 and debug flag, we should probably add a note about OS X. Something along those lines:

<b>Special Note for OS X:</b> Apple only provide a legacy profile (OpenGL 2.1) and a core profile for some versions of the operating system (further information are available on the <a href="https://developer.apple.com/opengl/capabilities/index.html">OpenGL Capabilities Tables</a> page). Moreover, no debug context are available on that system.

```
<b>Special Note for OS X:</b> Apple only provide a legacy profile (OpenGL 2.1) and a core profile for some versions of the operating system (further information are available on the <a href="https://developer.apple.com/opengl/capabilities/index.html">OpenGL Capabilities Tables</a> page). Moreover, no debug context are available on that system.
```